### ISSUE TITLE
Cannot authenticate after upgrading to 1.6.0

### ISSUE THREAD

**Issue by larswestman** [leeyuentuen/polestar_api#98]
Probably because Polestar has introduced MFA

**Comment by leeyuentuen** [comment]
which error did you see in the log

**Comment by warksit** [comment]
Just trying to re-add after changing my phone. I receive a "no cars found in account error"

````
This error originated from a custom integration.

Logger: custom_components.polestar_api.pypolestar.polestar
Source: /config/custom_components/polestar_api/pypolestar/polestar.py:112
Integration: Polestar API (documentation)
First occurred: 11:12:37 (3 occurrences)
Last logged: 11:19:37

No cars found in account
NoneType: None
````

**Comment by leeyuentuen** [comment]
if you login here

`https://polestarid.eu.polestar.com/as/authorization.oauth2?response_type=code&client_id=polmystar&redirect_uri=https%3A%2F%2Fwww.polestar.com%2Fsign-in-callback&scope=openid profile email customer:attributes customer:attributes:write
`

do you see there the car tab?

**Comment by warksit** [comment]
No. But you gave me an idea. It's my girlfriend's Polestar so she is the "owner" so I logged in to her account for your integration and that has worked. (I thought previously I had used my login so not sure is something has changed with delegated accounts)

**Comment by kristoferericsson** [comment]
Does not work for me either. I can se car stats on above link. But integration gets nothing. I removed it and tried to add again but now I get “ Unexpected error creating api.”

**Comment by kristoferericsson** [comment]
> > Does not work for me either. I can se car stats on above link. But integration gets nothing. I removed it and tried to add again but now I get “ Unexpected error creating api.”
> 
> check what for error you got

This error originated from a custom integration.

Logger: custom_components.polestar_api.pypolestar.polestar
Source: custom_components/polestar_api/pypolestar/polestar.py:193
Integration: Polestar API (documentation)
First occurred: January 18, 2024 at 22:08:27 (6 occurrences)
Last logged: 14:24:09

[{'message': 'Cannot return null for non-nullable field WorkOrder.status.', 'path': ['getConsumerCarsV2', 1, 'serviceHistory', 0, 'status'], 'extensions': {'code': 'INTERNAL_SERVER_ERROR', 'exception': {'stacktrace': ['Error: Cannot return null for non-nullable field WorkOrder.status.', ' at i2 (/var/task/src/functions/mystar/handler.js:83:5905)', ' at v4e (/var/task/src/functions/mystar/handler.js:83:5270)', ' at l5 (/var/task/src/functions/mystar/handler.js:83:4784)', ' at cue (/var/task/src/functions/mystar/handler.js:83:8691)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6202)', ' at /var/task/src/functions/mystar/handler.js:83:6630', ' at Function.from (<anonymous>)', ' at Kgr (/var/task/src/functions/mystar/handler.js:83:6515)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6060)', ' at v4e (/var/task/src/functions/mystar/handler.js:83:5270)', ' at l5 (/var/task/src/functions/mystar/handler.js:83:4784)', ' at cue (/var/task/src/functions/mystar/handler.js:83:8691)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6202)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:5864)', ' at /var/task/src/functions/mystar/handler.js:83:6630', ' at Function.from (<anonymous>)'], 'message': 'Cannot return null for non-nullable field WorkOrder.status.', 'locations': [{'line': 1, 'column': 482}], 'path': ['getConsumerCarsV2', 1, 'serviceHistory', 0, 'status']}, 'serviceName': 'mystar'}}, {'message': 'Cannot return null for non-nullable field WorkOrder.status.', 'path': ['getConsumerCarsV2', 1, 'serviceHistory', 1, 'status'], 'extensions': {'code': 'INTERNAL_SERVER_ERROR', 'exception': {'stacktrace': ['Error: Cannot return null for non-nullable field WorkOrder.status.', ' at i2 (/var/task/src/functions/mystar/handler.js:83:5905)', ' at v4e (/var/task/src/functions/mystar/handler.js:83:5270)', ' at l5 (/var/task/src/functions/mystar/handler.js:83:4784)', ' at cue (/var/task/src/functions/mystar/handler.js:83:8691)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6202)', ' at /var/task/src/functions/mystar/handler.js:83:6630', ' at Function.from (<anonymous>)', ' at Kgr (/var/task/src/functions/mystar/handler.js:83:6515)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6060)', ' at v4e (/var/task/src/functions/mystar/handler.js:83:5270)', ' at l5 (/var/task/src/functions/mystar/handler.js:83:4784)', ' at cue (/var/task/src/functions/mystar/handler.js:83:8691)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6202)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:5864)', ' at /var/task/src/functions/mystar/handler.js:83:6630', ' at Function.from (<anonymous>)'], 'message': 'Cannot return null for non-nullable field WorkOrder.status.', 'locations': [{'line': 1, 'column': 482}], 'path': ['getConsumerCarsV2', 1, 'serviceHistory', 1, 'status']}, 'serviceName': 'mystar'}}, {'message': 'Cannot return null for non-nullable field WorkOrder.status.', 'path': ['getConsumerCarsV2', 1, 'serviceHistory', 2, 'status'], 'extensions': {'code': 'INTERNAL_SERVER_ERROR', 'exception': {'stacktrace': ['Error: Cannot return null for non-nullable field WorkOrder.status.', ' at i2 (/var/task/src/functions/mystar/handler.js:83:5905)', ' at v4e (/var/task/src/functions/mystar/handler.js:83:5270)', ' at l5 (/var/task/src/functions/mystar/handler.js:83:4784)', ' at cue (/var/task/src/functions/mystar/handler.js:83:8691)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6202)', ' at /var/task/src/functions/mystar/handler.js:83:6630', ' at Function.from (<anonymous>)', ' at Kgr (/var/task/src/functions/mystar/handler.js:83:6515)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6060)', ' at v4e (/var/task/src/functions/mystar/handler.js:83:5270)', ' at l5 (/var/task/src/functions/mystar/handler.js:83:4784)', ' at cue (/var/task/src/functions/mystar/handler.js:83:8691)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6202)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:5864)', ' at /var/task/src/functions/mystar/handler.js:83:6630', ' at Function.from (<anonymous>)'], 'message': 'Cannot return null for non-nullable field WorkOrder.status.', 'locations': [{'line': 1, 'column': 482}], 'path': ['getConsumerCarsV2', 1, 'serviceHistory', 2, 'status']}, 'serviceName': 'mystar'}}, {'message': 'Cannot return null for non-nullable field WorkOrder.status.', 'path': ['getConsumerCarsV2', 1, 'serviceHistory', 3, 'status'], 'extensions': {'code': 'INTERNAL_SERVER_ERROR', 'exception': {'stacktrace': ['Error: Cannot return null for non-nullable field WorkOrder.status.', ' at i2 (/var/task/src/functions/mystar/handler.js:83:5905)', ' at v4e (/var/task/src/functions/mystar/handler.js:83:5270)', ' at l5 (/var/task/src/functions/mystar/handler.js:83:4784)', ' at cue (/var/task/src/functions/mystar/handler.js:83:8691)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6202)', ' at /var/task/src/functions/mystar/handler.js:83:6630', ' at Function.from (<anonymous>)', ' at Kgr (/var/task/src/functions/mystar/handler.js:83:6515)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6060)', ' at v4e (/var/task/src/functions/mystar/handler.js:83:5270)', ' at l5 (/var/task/src/functions/mystar/handler.js:83:4784)', ' at cue (/var/task/src/functions/mystar/handler.js:83:8691)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6202)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:5864)', ' at /var/task/src/functions/mystar/handler.js:83:6630', ' at Function.from (<anonymous>)'], 'message': 'Cannot return null for non-nullable field WorkOrder.status.', 'locations': [{'line': 1, 'column': 482}], 'path': ['getConsumerCarsV2', 1, 'serviceHistory', 3, 'status']}, 'serviceName': 'mystar'}}, {'message': 'Cannot return null for non-nullable field WorkOrder.status.', 'path': ['getConsumerCarsV2', 1, 'serviceHistory', 4, 'status'], 'extensions': {'code': 'INTERNAL_SERVER_ERROR', 'exception': {'stacktrace': ['Error: Cannot return null for non-nullable field WorkOrder.status.', ' at i2 (/var/task/src/functions/mystar/handler.js:83:5905)', ' at v4e (/var/task/src/functions/mystar/handler.js:83:5270)', ' at l5 (/var/task/src/functions/mystar/handler.js:83:4784)', ' at cue (/var/task/src/functions/mystar/handler.js:83:8691)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6202)', ' at /var/task/src/functions/mystar/handler.js:83:6630', ' at Function.from (<anonymous>)', ' at Kgr (/var/task/src/functions/mystar/handler.js:83:6515)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6060)', ' at v4e (/var/task/src/functions/mystar/handler.js:83:5270)', ' at l5 (/var/task/src/functions/mystar/handler.js:83:4784)', ' at cue (/var/task/src/functions/mystar/handler.js:83:8691)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6202)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:5864)', ' at /var/task/src/functions/mystar/handler.js:83:6630', ' at Function.from (<anonymous>)'], 'message': 'Cannot return null for non-nullable field WorkOrder.status.', 'locations': [{'line': 1, 'column': 482}], 'path': ['getConsumerCarsV2', 1, 'serviceHistory', 4, 'status']}, 'serviceName': 'mystar'}}, {'message': 'Cannot return null for non-nullable field WorkOrder.status.', 'path': ['getConsumerCarsV2', 1, 'serviceHistory', 5, 'status'], 'extensions': {'code': 'INTERNAL_SERVER_ERROR', 'exception': {'stacktrace': ['Error: Cannot return null for non-nullable field WorkOrder.status.', ' at i2 (/var/task/src/functions/mystar/handler.js:83:5905)', ' at v4e (/var/task/src/functions/mystar/handler.js:83:5270)', ' at l5 (/var/task/src/functions/mystar/handler.js:83:4784)', ' at cue (/var/task/src/functions/mystar/handler.js:83:8691)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6202)', ' at /var/task/src/functions/mystar/handler.js:83:6630', ' at Function.from (<anonymous>)', ' at Kgr (/var/task/src/functions/mystar/handler.js:83:6515)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6060)', ' at v4e (/var/task/src/functions/mystar/handler.js:83:5270)', ' at l5 (/var/task/src/functions/mystar/handler.js:83:4784)', ' at cue (/var/task/src/functions/mystar/handler.js:83:8691)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6202)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:5864)', ' at /var/task/src/functions/mystar/handler.js:83:6630', ' at Function.from (<anonymous>)'], 'message': 'Cannot return null for non-nullable field WorkOrder.status.', 'locations': [{'line': 1, 'column': 482}], 'path': ['getConsumerCarsV2', 1, 'serviceHistory', 5, 'status']}, 'serviceName': 'mystar'}}, {'message': 'Cannot return null for non-nullable field WorkOrder.status.', 'path': ['getConsumerCarsV2', 1, 'serviceHistory', 6, 'status'], 'extensions': {'code': 'INTERNAL_SERVER_ERROR', 'exception': {'stacktrace': ['Error: Cannot return null for non-nullable field WorkOrder.status.', ' at i2 (/var/task/src/functions/mystar/handler.js:83:5905)', ' at v4e (/var/task/src/functions/mystar/handler.js:83:5270)', ' at l5 (/var/task/src/functions/mystar/handler.js:83:4784)', ' at cue (/var/task/src/functions/mystar/handler.js:83:8691)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6202)', ' at /var/task/src/functions/mystar/handler.js:83:6630', ' at Function.from (<anonymous>)', ' at Kgr (/var/task/src/functions/mystar/handler.js:83:6515)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6060)', ' at v4e (/var/task/src/functions/mystar/handler.js:83:5270)', ' at l5 (/var/task/src/functions/mystar/handler.js:83:4784)', ' at cue (/var/task/src/functions/mystar/handler.js:83:8691)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6202)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:5864)', ' at /var/task/src/functions/mystar/handler.js:83:6630', ' at Function.from (<anonymous>)'], 'message': 'Cannot return null for non-nullable field WorkOrder.status.', 'locations': [{'line': 1, 'column': 482}], 'path': ['getConsumerCarsV2', 1, 'serviceHistory', 6, 'status']}, 'serviceName': 'mystar'}}, {'message': 'Cannot return null for non-nullable field WorkOrder.status.', 'path': ['getConsumerCarsV2', 1, 'serviceHistory', 7, 'status'], 'extensions': {'code': 'INTERNAL_SERVER_ERROR', 'exception': {'stacktrace': ['Error: Cannot return null for non-nullable field WorkOrder.status.', ' at i2 (/var/task/src/functions/mystar/handler.js:83:5905)', ' at v4e (/var/task/src/functions/mystar/handler.js:83:5270)', ' at l5 (/var/task/src/functions/mystar/handler.js:83:4784)', ' at cue (/var/task/src/functions/mystar/handler.js:83:8691)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6202)', ' at /var/task/src/functions/mystar/handler.js:83:6630', ' at Function.from (<anonymous>)', ' at Kgr (/var/task/src/functions/mystar/handler.js:83:6515)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6060)', ' at v4e (/var/task/src/functions/mystar/handler.js:83:5270)', ' at l5 (/var/task/src/functions/mystar/handler.js:83:4784)', ' at cue (/var/task/src/functions/mystar/handler.js:83:8691)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6202)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:5864)', ' at /var/task/src/functions/mystar/handler.js:83:6630', ' at Function.from (<anonymous>)'], 'message': 'Cannot return null for non-nullable field WorkOrder.status.', 'locations': [{'line': 1, 'column': 482}], 'path': ['getConsumerCarsV2', 1, 'serviceHistory', 7, 'status']}, 'serviceName': 'mystar'}}, {'message': 'Cannot return null for non-nullable field WorkOrder.status.', 'path': ['getConsumerCarsV2', 1, 'serviceHistory', 8, 'status'], 'extensions': {'code': 'INTERNAL_SERVER_ERROR', 'exception': {'stacktrace': ['Error: Cannot return null for non-nullable field WorkOrder.status.', ' at i2 (/var/task/src/functions/mystar/handler.js:83:5905)', ' at v4e (/var/task/src/functions/mystar/handler.js:83:5270)', ' at l5 (/var/task/src/functions/mystar/handler.js:83:4784)', ' at cue (/var/task/src/functions/mystar/handler.js:83:8691)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6202)', ' at /var/task/src/functions/mystar/handler.js:83:6630', ' at Function.from (<anonymous>)', ' at Kgr (/var/task/src/functions/mystar/handler.js:83:6515)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6060)', ' at v4e (/var/task/src/functions/mystar/handler.js:83:5270)', ' at l5 (/var/task/src/functions/mystar/handler.js:83:4784)', ' at cue (/var/task/src/functions/mystar/handler.js:83:8691)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6202)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:5864)', ' at /var/task/src/functions/mystar/handler.js:83:6630', ' at Function.from (<anonymous>)'], 'message': 'Cannot return null for non-nullable field WorkOrder.status.', 'locations': [{'line': 1, 'column': 482}], 'path': ['getConsumerCarsV2', 1, 'serviceHistory', 8, 'status']}, 'serviceName': 'mystar'}}, {'message': 'Cannot return null for non-nullable field WorkOrder.status.', 'path': ['getConsumerCarsV2', 1, 'serviceHistory', 9, 'status'], 'extensions': {'code': 'INTERNAL_SERVER_ERROR', 'exception': {'stacktrace': ['Error: Cannot return null for non-nullable field WorkOrder.status.', ' at i2 (/var/task/src/functions/mystar/handler.js:83:5905)', ' at v4e (/var/task/src/functions/mystar/handler.js:83:5270)', ' at l5 (/var/task/src/functions/mystar/handler.js:83:4784)', ' at cue (/var/task/src/functions/mystar/handler.js:83:8691)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6202)', ' at /var/task/src/functions/mystar/handler.js:83:6630', ' at Function.from (<anonymous>)', ' at Kgr (/var/task/src/functions/mystar/handler.js:83:6515)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6060)', ' at v4e (/var/task/src/functions/mystar/handler.js:83:5270)', ' at l5 (/var/task/src/functions/mystar/handler.js:83:4784)', ' at cue (/var/task/src/functions/mystar/handler.js:83:8691)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6202)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:5864)', ' at /var/task/src/functions/mystar/handler.js:83:6630', ' at Function.from (<anonymous>)'], 'message': 'Cannot return null for non-nullable field WorkOrder.status.', 'locations': [{'line': 1, 'column': 482}], 'path': ['getConsumerCarsV2', 1, 'serviceHistory', 9, 'status']}, 'serviceName': 'mystar'}}, {'message': 'Cannot return null for non-nullable field WorkOrder.status.', 'path': ['getConsumerCarsV2', 1, 'serviceHistory', 10, 'status'], 'extensions': {'code': 'INTERNAL_SERVER_ERROR', 'exception': {'stacktrace': ['Error: Cannot return null for non-nullable field WorkOrder.status.', ' at i2 (/var/task/src/functions/mystar/handler.js:83:5905)', ' at v4e (/var/task/src/functions/mystar/handler.js:83:5270)', ' at l5 (/var/task/src/functions/mystar/handler.js:83:4784)', ' at cue (/var/task/src/functions/mystar/handler.js:83:8691)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6202)', ' at /var/task/src/functions/mystar/handler.js:83:6630', ' at Function.from (<anonymous>)', ' at Kgr (/var/task/src/functions/mystar/handler.js:83:6515)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6060)', ' at v4e (/var/task/src/functions/mystar/handler.js:83:5270)', ' at l5 (/var/task/src/functions/mystar/handler.js:83:4784)', ' at cue (/var/task/src/functions/mystar/handler.js:83:8691)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6202)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:5864)', ' at /var/task/src/functions/mystar/handler.js:83:6630', ' at Function.from (<anonymous>)'], 'message': 'Cannot return null for non-nullable field WorkOrder.status.', 'locations': [{'line': 1, 'column': 482}], 'path': ['getConsumerCarsV2', 1, 'serviceHistory', 10, 'status']}, 'serviceName': 'mystar'}}, {'message': 'Cannot return null for non-nullable field WorkOrder.status.', 'path': ['getConsumerCarsV2', 1, 'serviceHistory', 11, 'status'], 'extensions': {'code': 'INTERNAL_SERVER_ERROR', 'exception': {'stacktrace': ['Error: Cannot return null for non-nullable field WorkOrder.status.', ' at i2 (/var/task/src/functions/mystar/handler.js:83:5905)', ' at v4e (/var/task/src/functions/mystar/handler.js:83:5270)', ' at l5 (/var/task/src/functions/mystar/handler.js:83:4784)', ' at cue (/var/task/src/functions/mystar/handler.js:83:8691)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6202)', ' at /var/task/src/functions/mystar/handler.js:83:6630', ' at Function.from (<anonymous>)', ' at Kgr (/var/task/src/functions/mystar/handler.js:83:6515)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6060)', ' at v4e (/var/task/src/functions/mystar/handler.js:83:5270)', ' at l5 (/var/task/src/functions/mystar/handler.js:83:4784)', ' at cue (/var/task/src/functions/mystar/handler.js:83:8691)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:6202)', ' at i2 (/var/task/src/functions/mystar/handler.js:83:5864)', ' at /var/task/src/functions/mystar/handler.js:83:6630', ' at Function.from (<anonymous>)'], 'message': 'Cannot return null for non-nullable field WorkOrder.status.', 'locations': [{'line': 1, 'column': 482}], 'path': ['getConsumerCarsV2', 1, 'serviceHistory', 11, 'status']}, 'serviceName': 'mystar'}}]

**Comment by leeyuentuen** [comment]
i have put extra field for all the request call. it seems it failed here. in the workorder parts (that we don't use them atm) i'll later take a look how the fix them

**Comment by Depechie** [comment]
I get : No token in response
Same credentials work for logging into the site

**Comment by leeyuentuen** [comment]
@kristoferericsson  can you take a look at this beta if it solved your issue?
https://github.com/leeyuentuen/polestar_api/releases/tag/1.6.1

**Comment by kristoferericsson** [comment]
> @kristoferericsson can you take a look at this beta if it solved your issue? https://github.com/leeyuentuen/polestar_api/releases/tag/1.6.1

@leeyuentuen it worked fine now!

**Comment by leeyuentuen** [comment]
> I get : No token in response Same credentials work for logging into the site

@Depechie maybe you fill incorrect credential? check in log what kind of error you get

**Comment by Depechie** [comment]
> > I get : No token in response Same credentials work for logging into the site
> 
> @Depechie maybe you fill incorrect credential? check in log what kind of error you get

FFS there was a space at the end of the password all is well now

**Comment by leeyuentuen** [comment]
@larswestman can you provide the log with error?

atm i didn't got mfa request

**Comment by leeyuentuen** [comment]
not an issue atm with mfa, but  maybe other issue with workorder, has been solved now on the release 1.6.3
